File: java/src/main/java/org/xllgl2017/Cookie.java
package org.xllgl2017;

public class Cookie {
    private final String name;
    private final String value;
    private int age;
    private String domain;
    private String path;
    private boolean httpOnly;
    private boolean secure;
    private String expires;
    private String sameSite;
    private boolean icpsp;

    public Cookie(String name, String value) {
        this.name = name;
        this.value = value;
    }

    public String getValue() {
        return value;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    public String getDomain() {
        return domain;
    }

    public String getPath() {
        return path;
    }

    public boolean isHttpOnly() {
        return httpOnly;
    }

    public boolean isSecure() {
        return secure;
    }

    public String getExpires() {
        return expires;
    }

    public String getSameSite() {
        return sameSite;
    }

    public boolean isIcpsp() {
        return icpsp;
    }
}

File: qt/Cookie.cpp
//
// Created by XLX on 2026/1/21.
//

#include "Cookie.h"

Cookie::Cookie() = default;

Cookie::Cookie(QString name, QString value) {
    this->name = std::move(name);
    this->value = std::move(value);
}

Cookie::Cookie(const QJsonObject& cookie) {
    this->name = cookie.value("name").toString();
    this->value = cookie.value("value").toString();
    this->age = cookie.value("age").toInt();
    this->domain = cookie.value("domain").toString();
    this->path = cookie.value("path").toString();
    this->httpOnly = cookie.value("http_only").toBool();
    this->secure = cookie.value("secure").toBool();
    this->expires = cookie.value("expires").toString();
    this->sameSite = cookie.value("same_site").toString();
    this->icpsp = cookie.value("icpsp").toBool();
}

QString Cookie::getName() {
    return this->name;
}

QString Cookie::getValue() {
    return this->value;
}
